rules/TypeDeclaration/Rector/ClassMethod/NumericReturnTypeFromStrictScalarReturnsRector.php

```diff
@@ -141,7 +153,7 @@ private function refactorBinaryOp(
         $leftType = $this->getType($binaryOp->left);
         $rightType = $this->getType($binaryOp->right);
 
-        if ($leftType instanceof IntegerType && $rightType instanceof IntegerType) {
+        if ($leftType->isInteger()->yes() && $rightType->isInteger()->yes()) {
```

Comment:
how should this rector behave according to phpdoc typed params?

does the "strict" in the rector name mean it should only trust native types?

Reply:
I am thinking about a test like

```
<?php

namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\NumericReturnTypeFromStrictScalarReturnsRector\Fixture;

final class SkipPhpDoc
{
    /**
     * @param 0|1 $first
     */
    public function resolve($first, int $second)
    {
        return $first + $second;
    }
}

```

Reply:
Docblocks are skipped as non-strict types.

Reply:
Could you add test fixture with this code sample to make sure its still skipped?

Reply:
There is ExprAnalyzer for detect variable from non typed param for that

https://github.com/rectorphp/rector-src/blob/93a279d3658fea5d56670bb63ff41df3b2b481f3/src/NodeAnalyzer/ExprAnalyzer.php#L36

Reply:
@staabm could you verify latest change of `ExprAnalyzer->isNonTypedFromParam()` works on this? Thank you.

Reply:
it seems it works - thank you